firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -329,4 +328,15 @@ private String getFingerprintHashForPackage() {
       return null;
     }
   }
+
+  /**
+   * Returns parsed token expiration timestamp in seconds.
+   *
+   * @param expiresIn is expiration timestamp in String format: 604800s
+   */
+  private static long parseTokenExpirationTimestamp(String expiresIn) {
+    return (expiresIn == null || expiresIn.length() == 0)
+        ? 0L
+        : Long.parseLong(expiresIn.substring(0, expiresIn.length() - 1));
```

Comment:
I recommend to do a regex check on the string, which pattern is [0-9]+s

Reply:
Updated. PTAL.